src/sphinxcontrib/bibtex/directives.py

```diff
@@ -97,21 +97,25 @@ def run(self):
         if "filter" in self.options:
             if "all" in self.options:
                 logger.warning(":filter: overrides :all:",
-                               location=(env.docname, self.lineno))
+                               location=(env.docname, self.lineno),
+                               type="bibtex", subtype="filter_overrides_all")
             if "notcited" in self.options:
                 logger.warning(":filter: overrides :notcited:",
-                               location=(env.docname, self.lineno))
+                               location=(env.docname, self.lineno),
+                               type="bibtex", subtype="filter_overrides_notcited")
             if "cited" in self.options:
                 logger.warning(":filter: overrides :cited:",
-                               location=(env.docname, self.lineno))
+                               location=(env.docname, self.lineno),
+                               type="bibtex", subtype="filter_overrides_cited")
             try:
                 filter_ = ast.parse(self.options["filter"])
             except SyntaxError:
                 logger.warning(
                     "syntax error in :filter: expression" +
                     " (" + self.options["filter"] + "); "
                     "the option will be ignored",
-                    location=(env.docname, self.lineno)
+                    location=(env.docname, self.lineno,
+                              type="bibtex", subtype="filter_syntax_error")
```

Comment:
The type and subtype argument should be direct arguments of the warning call here, and not be part of the location tuple.